packages/big-design/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1,200 +1,183 @@
-        <List
-          {...rest}
-          aria-labelledby={this.getTriggerId()}
-          handleListRef={this.listRef}
-          id={this.getDropdownId()}
-          isOpen={isOpen}
-          maxHeight={maxHeight}
-          onKeyDown={this.handleOnDropdownKeyDown}
-          placement={placement}
-          role="listbox"
-          {...aria}
-        >
-          {this.renderItems()}
-        </List>
-      </Manager>
+      <Popper placement={placement} modifiers={popperModifiers} eventsEnabled={isOpen}>
+        {({ placement: popperPlacement, ref, scheduleUpdate, style: popperStyle }) => (
+          <List
+            {...rest}
+            {...getMenuProps({ ref })}
+            data-placement={popperPlacement}
+            isOpen={isOpen}
+            maxHeight={maxHeight}
+            scheduleUpdate={scheduleUpdate}
+            style={popperStyle}
+          >
+            {renderChildren()}
+          </List>
+        )}
+      </Popper>
     );
   }
 
-  private isGroup(item: DropdownOption<T> | DropdownOptionGroup<T>) {
-    return 'options' in item && !('content' in item);
+  function isGroup(item: DropdownItem<T> | DropdownItemGroup<T>) {
+    return 'items' in item && !('content' in item);
   }
 
-  private isOption(item: DropdownOption<T> | DropdownOptionGroup<T>) {
+  function isItem(item: DropdownItem<T> | DropdownItemGroup<T>) {
     return 'content' in item && !('options' in item);
   }
 
-  private renderItems() {
-    const { options } = this.props;
+  function renderChildren() {
+    // reset the key every time we rerender children
+    key = -1;
 
-    if (Array.isArray(options) && options.every(this.isGroup)) {
-      return (options as Array<DropdownOptionGroup<T>>).map((group, groupIndex) => this.renderGroup(group, groupIndex));
+    if (Array.isArray(items) && items.every(isGroup)) {
+      return (items as Array<DropdownItemGroup<T>>).map((group, groupIndex) => renderGroup(group, groupIndex));
     }
 
-    if (Array.isArray(options) && options.every(this.isOption)) {
-      return this.renderOptions(options as Array<DropdownOption<T>>);
+    if (Array.isArray(items) && items.every(isItem)) {
+      return renderItems(items as Array<DropdownItem<T>>);
     }
 
     return;
   }
 
-  private renderOptions(options: Array<DropdownOption<T>>, groupIndex: number | null = null) {
-    const { highlightedItem } = this.state;
-
+  function renderGroup(group: DropdownItemGroup<T>, groupIndex: number) {
     return (
-      Array.isArray(options) &&
-      options.map((option, index) => {
-        if (!option.content) {
-          return null;
-        }
-
-        const id = this.getItemId(option, index, groupIndex);
-        const isHighlighted = Boolean(highlightedItem && id === highlightedItem.id);
-        const ref = React.createRef<HTMLLIElement>();
-
-        if (!option.disabled) {
-          this.listItemsRefs.push(ref);
-        }
+      <Fragment key={groupIndex}>
```

Comment:
`key` can't be specified here, it has to be on the `map`.

Reply:
`groupIndex` comes from the map.

Reply:
https://reactjs.org/docs/lists-and-keys.html#extracting-components-with-keys